spyder/config/main.py

```diff
@@ -308,6 +308,8 @@
             ('pylint',
              {
               'enable': True,
+              'max_entries': 30,
+              'history_filenames': [],

```

Comment:
Please remove this option from here and add it instead to the `NAME_MAP` dict below, by creating a key under `transient` with this format:

```python
('pylint', [
    'history_filenames',
  ]
)
```

The thing is this option is transient because it depends on file names passed by the user to the Code Analysis combobox. So there's no need to register it here.

Reply:
Yep, sounds very reasonable.